Source/JavaScriptCore/wasm/WasmLLIntGenerator.cpp

```diff
@@ -1428,8 +1428,10 @@ auto LLIntGenerator::addBranchNull(ControlType& data, ExpressionType reference,
     // Leave a hole for the reference and avoid overwriting it with the condition.
     if (!shouldNegate)
         push();
+    else
+        checkConsistency(); // Explicitly check consistency even if we don't have a result.
 
-    auto condition = push();
+    auto condition = push(NoConsistencyCheck);
     WasmRefIsNull::emit(this, condition, reference);
```

Comment:
Since this push is only for the purpose of this null check, let's `checkConsistency()` after the null check to ensure that the stack has been re-adjusted back.

Reply:
Discussed offline. The consistency should be preserved after the stack size adjustment below in that function.
```
m_stackSize -= (1 + (shouldNegate ? 0 : 1));
```